### PR TITLE
chore: give context to errors (#10592)

### PR DESCRIPTION
**Pull request by vipin-m-s**
#10592 

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: util/settings/settings.go [hunk 1]
```diff
@@ -759,13 +759,13 @@ func (mgr *SettingsManager) GetServerRBACLogEnforceEnable() (bool, error) {
 func (mgr *SettingsManager) GetDeepLinks(deeplinkType string) ([]DeepLink, error) {
 	argoCDCM, err := mgr.getConfigMap()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error retrieving deep links %w", err)
```
Comment (crenshaw-dev) [inline util/settings/settings.go:762]:
```suggestion
		return nil, fmt.Errorf("error retrieving argocd-cm: %w", err)
```

The outer wrap will include the information that this is related to deep links.

File: util/settings/settings.go [hunk 2]
```diff
@@ -1036,7 +1036,7 @@ func (mgr *SettingsManager) GetHelmSettings() (*v1alpha1.HelmOptions, error) {
 func (mgr *SettingsManager) GetKustomizeSettings() (*KustomizeSettings, error) {
 	argoCDCM, err := mgr.getConfigMap()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error retrieving config map %w", err)
```
Comment (crenshaw-dev) [inline util/settings/settings.go:1039]:
```suggestion
		return nil, fmt.Errorf("error retrieving argocd-cm: %w", err)
```

For consistency

File: util/settings/settings.go [hunk 3]
```diff
@@ -688,14 +688,14 @@ func (mgr *SettingsManager) GetConfigMapByName(configMapName string) (*apiv1.Con
 func (mgr *SettingsManager) GetResourcesFilter() (*ResourcesFilter, error) {
 	argoCDCM, err := mgr.getConfigMap()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error retrieving Argo CD config map %w", err)
```
Comment (crenshaw-dev) [inline util/settings/settings.go:691]:
```suggestion
		return nil, fmt.Errorf("error retrieving argocd-cm: %w", err)
```

File: util/tls/tls.go [hunk 4]
```diff
@@ -123,11 +123,11 @@ func tlsVersionsToStr(versions []uint16) []string {
 func getTLSConfigCustomizer(minVersionStr, maxVersionStr, tlsCiphersStr string) (ConfigCustomizer, error) {
 	minVersion, err := getTLSVersionByString(minVersionStr)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error retrieving TLS version by string %s", err)
```
Comment (crenshaw-dev) [inline util/tls/tls.go:126]:
```suggestion
		return nil, fmt.Errorf("error retrieving TLS version by min version %q: %q", minVersionStr, err)
```

File: util/tls/tls.go [hunk 5]
```diff
@@ -123,11 +123,11 @@ func tlsVersionsToStr(versions []uint16) []string {
...
 	}
 	maxVersion, err := getTLSVersionByString(maxVersionStr)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error retrieving TLS version by string %s", err)
```
Comment (crenshaw-dev) [inline util/tls/tls.go:130]:
```suggestion
		return nil, fmt.Errorf("error retrieving TLS version by max version %q: %q", maxVersionStr, err)
```

File: util/tls/tls.go [hunk 6]
```diff
@@ -153,7 +153,7 @@ func getTLSConfigCustomizer(minVersionStr, maxVersionStr, tlsCiphersStr string)
 	if tlsCiphersStr != "" {
 		cipherSuites, err = getTLSCipherSuitesByString(tlsCiphersStr)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("error retrieving TLS cipher suites by string %w", err)
```
Comment (crenshaw-dev) [inline util/tls/tls.go:156]:
```suggestion
			return nil, fmt.Errorf("error retrieving TLS cipher suites: %w", err)
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Requested small tweaks. Also, please add `: ` before each `%w`. It's just the convention, to separate messages. :-)

**Comment by vipinachar** [comment]
Thank you  for reviewing , I will follow the convention going forward. : )

File: util/settings/settings.go [hunk 7]
```diff
@@ -2107,14 +2107,14 @@ func ReplaceStringSecret(val string, secretValues map[string]string) string {
...
-		return nil, err
+		return nil, fmt.Errorf("error retrieving argocd-cm: %w", err)
 	}
 	globalProjectSettings := make([]GlobalProjectSettings, 0)
 	if value, ok := argoCDCM.Data[globalProjectsKey]; ok {
 		if value != "" {
 			err := yaml.Unmarshal([]byte(value), &globalProjectSettings)
 			if err != nil {
-				return nil, err
+				return nil, fmt.Errorf("error unmarshalling global project settings %w", err)
```
Comment (crenshaw-dev) [inline util/settings/settings.go:2117]:
```suggestion
				return nil, fmt.Errorf("error unmarshalling global project settings: %w", err)
```

File: util/settings/settings.go [hunk 8]
```diff
@@ -1238,15 +1238,15 @@ func (mgr *SettingsManager) GetSettings() (*ArgoCDSettings, error) {
 	}
 	argoCDCM, err := mgr.configmaps.ConfigMaps(mgr.namespace).Get(common.ArgoCDConfigMapName)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error retrieving Argo CD config map: %w", err)
```
Comment (crenshaw-dev) [inline util/settings/settings.go:1241]:
```suggestion
		return nil, fmt.Errorf("error retrieving argocd-cm: %w", err)
```

File: util/tls/tls.go [hunk 9]
```diff
@@ -123,11 +123,11 @@ func tlsVersionsToStr(versions []uint16) []string {
 func getTLSConfigCustomizer(minVersionStr, maxVersionStr, tlsCiphersStr string) (ConfigCustomizer, error) {
 	minVersion, err := getTLSVersionByString(minVersionStr)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error retrieving TLS version by min version %q: %q", minVersionStr, err)
```
Comment (crenshaw-dev) [inline util/tls/tls.go:126]:
```suggestion
		return nil, fmt.Errorf("error retrieving TLS version by min version %q: %w", minVersionStr, err)
```

File: util/tls/tls.go [hunk 10]
```diff
@@ -123,11 +123,11 @@ func tlsVersionsToStr(versions []uint16) []string {
...
 	}
 	maxVersion, err := getTLSVersionByString(maxVersionStr)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error retrieving TLS version by max version %q: %q", maxVersionStr, err)
```
Comment (crenshaw-dev) [inline util/tls/tls.go:130]:
```suggestion
		return nil, fmt.Errorf("error retrieving TLS version by max version %q: %w", maxVersionStr, err)
```

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**